Fixes #137 AWS sdk v3 upgrade

saravanak opened the pull request:
When building an Ember app using the addon, we see this message:

`NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
Please migrate your code to use AWS SDK for JavaScript (v3).`

This PR upgrades the aws-js sdk to v3. 
Ref: https://www.npmjs.com/package/aws-sdk-js-codemod Applied the transform from the above codemod

## What Changed & Why
Upgraded AWS SDK version from 2 to 3. 

## Related issues
 #137  

## PR Checklist
- [x] Made sure all tests pass

## People
@lukemelia  @mattstrayer

Commit:
Add asdf tool-versions file to the gitignore

Commit:
Upgrade to use AWS SDK v3.

Ref: https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/migrating-to-v3.html

lukemelia commented:
@saravanak looks lie CI is failing. Can you take a look?

Commit:
Fix tests

Assume that the provided `s3Client` conforms to the v3 API to make tests
pass.

Commit:
Add CHANGELOG referencing the PR for AWS Upgrade

Commit:
v4.0.0-0

Commit:
Update documentation

Commit:
Revert "Add CHANGELOG referencing the PR for AWS Upgrade"

This reverts commit d87beb43662f4fc150948ecc93ea4a7c8d4a2a57.

(This will be done by release-it flow)

saravanak commented:
> @saravanak looks lie CI is failing. Can you take a look?

@lukemelia  I have fixed the tests by 

 - Removing references to `aws-sdk` library in both tests and source. I have also assumed that the `s3Client` that, if passed,  should conform to the v3 API of the sdk. I think this might need a semantic major version update, but only if the tests pass and you are happy with the changes. 

PS: I will work on https://github.com/ember-cli-deploy/ember-cli-deploy-s3/pull/183 after we are done with the conversation here.

lukemelia commented:
@saravanak Looks good, and I agree with your choice regarding the s3client config. I also agree that this should be a major, with a CHANGELOG explanation of what the breaking change is. The easiest way to do this is to update the description of this PR to include that information.

saravanak commented:
@lukemelia I have 
 - Created a new major release with a pre-release tag on it. We can test it on our projects after the release and then we can promote it 
 - Added relevant documentation in Readme.md 
 - Added changelog entry 
 
I see from the Release.md file that we need to create proper labels for the PRs. Seems like I don't have access to do it myself. Assuming you will do it.
